Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -194,7 +217,11 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
   }
 
   fileprivate let sortPagerSelectedSortProperty = MutableProperty<DiscoveryParams.Sort?>(nil)
+
+  // Stores last page before new page transition, default page is .magic, because it's the first page on Application Opened
+  fileprivate let previouslySortPagerSelectedSortProperty = MutableProperty<DiscoveryParams.Sort>(.magic)
```

Comment:
I see that you've used the property name above to name this but this does not read well at all. This is a very long, confusing name. I also don't think you should be defaulting given the other properties.

Reply:
I'll revisit this, but the default seem to be need, when the app loaded the default sorted is magic, at the point where the function is called, `previouslySortPagerSelectedSortProperty ` can't be nil.